python/lsst/daf/butler/registry/_registry.py

```diff
@@ -1032,6 +1033,72 @@ def insertDatasetLocations(self, datastoreName: str, refs: Iterable[DatasetRef])
             *[{"datastore_name": datastoreName, "dataset_id": _checkAndGetId(ref)} for ref in refs]
         )
 
+    @transactional
+    def moveDatasetLocationToTrash(self, datastoreName: str, refs: Iterable[DatasetRef]):
+        """Move the dataset location information to trash.
+
+        Parameters
+        ----------
+        datastoreName : `str`
+            Name of the datastore holding these datasets.
+        refs : `~collections.abc.Iterable` of `DatasetRef`
+            References to the datasets.
+        """
+        # We only want to move rows that already exist in the main table
+        filtered = self.checkDatasetLocations(datastoreName, refs)
+        self.canDeleteDatasetLocations(datastoreName, filtered)
```

Comment:
I'm fine with you taking another stab at this. The reason I had to do it was that in some cases a DatasetRef can refer to components that have already been trashed (since we can't turn the component off within the DatasetRef) so switching things to only moving rows that are still there prevented me from adding rows to the trash that were already listed in the trash.